### ISSUE TITLE
service_rngd_enabled rule is failing after reboot

### ISSUE THREAD

**Issue by marcusburghardt** [ComplianceAsCode/content#10111]
#### Description of problem:

The `service_rngd_enabled` rule is failing after reboot when the system is scanned and remediated using
`stig` profile.

#### SCAP Security Guide Version:

master as of 2023-01-21

#### Operating System Version:

RHEL8 and RHEL9

#### Steps to Reproduce:

1. Scan and remediate the system using `stig` profile.
2. Reboot the system.
3. Scan the system again after reboot.
4. The `xccdf_org.ssgproject.content_rule_service_rngd_enabled` rule will fail.

#### Actual Results:

The `xccdf_org.ssgproject.content_rule_service_rngd_enabled` rule fails.

#### Expected Results:

The `xccdf_org.ssgproject.content_rule_service_rngd_enabled` rule pass.

#### Additional Information/Debugging Steps:

Only manifested in `s390x` architecture.

**Comment by marcusburghardt** [comment]
I just saw there was another issue related to this: https://github.com/ComplianceAsCode/content/issues/10080

**Comment by jan-cerny** [comment]
> I just saw there was another issue related to this: https://github.com/ComplianceAsCode/content/issues/10080

and that one links a BZ https://bugzilla.redhat.com/show_bug.cgi?id=2156554 that suggests that it's a problem in the service itself

**Comment by comps** [comment]
> > I just saw there was another issue related to this: #10080
> 
> and that one links a BZ https://bugzilla.redhat.com/show_bug.cgi?id=2156554 that suggests that it's a problem in the service itself

Yes, it seems that the downstream RHEL maintainer of the `rng-tools` package has added

```diff
diff --git a/rngd.service b/rngd.service
index 0cf8106..911cec1 100644
--- a/rngd.service
+++ b/rngd.service
@@ -1,6 +1,7 @@
 [Unit]
 Description=Hardware RNG Entropy Gatherer Daemon
 ConditionVirtualization=!container
+ConditionKernelCommandLine=!fips=1
 
 # The "-f" option is required for the systemd service rngd to work with Type=simple
 [Service]
```
in (at least) RHEL-8.8, so that `rngd` never starts on a FIPS-enabled system, which includes STIG.

That's why the rule fails now, even with (supposedly fixed) `rng-tools-6.15-3.el8`.

...
Note that it might be wise to take this opportunity and get rid of the rule completely - there should be no real need for `rngd` on a modern Fedora or RHEL system (or any Linux for that matter), except in very specific cases (entropy research and measurement or fixing hardware-specific RNG issues), see https://bugzilla.redhat.com/show_bug.cgi?id=2028720 for some of the analysis (there are several other, older bugs too).

**Comment by mildas** [comment]
The problem with FIPS occurs also on RHEL9.
`rng-tools-6.15-3.el9` was tested in last test run and `rngd` still doesn't start. The failure is on all architectures.
